From 0cef611a14c640a2436ec96ce8211e8fc950612a Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Tue, 20 Nov 2018 16:35:49 +0530
Subject: [PATCH 0760/1051] soc: octeontx2: Fix SSO HWGRP configuration

Trying to drain NPA buffer pointers from SSO cache
even before mapping NPA PF_FUNC to SSO group is useless.
Moreover current code doesn't even set
SSO_AF_HWGRPX_AW_CFG[XAQ_ALLOC_DIS] but still polls on
SSO_AF_HWGRPX_AW_STATUS[XAQ_BUF_CACHED] to return zero.

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_sso.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_sso.c b/drivers/soc/marvell/octeontx2/rvu_sso.c
index ea7ecc3acded..44b8ed5c85fe 100644
--- a/drivers/soc/marvell/octeontx2/rvu_sso.c
+++ b/drivers/soc/marvell/octeontx2/rvu_sso.c
@@ -196,18 +196,6 @@ int rvu_mbox_handler_sso_hw_setconfig(struct rvu *rvu,
 		if (lf < 0)
 			return SSO_AF_ERR_LF_INVALID;
 
-		/* Disable and drain previous config */
-		rvu_write64(rvu, blkaddr, SSO_AF_HWGRPX_AW_CFG(lf), 0x0);
-
-		err = rvu_poll_reg(rvu, blkaddr,
-				   SSO_AF_HWGRPX_AW_STATUS(lf),
-				   SSO_HWGRP_AW_STS_XAQ_BUFSC_MASK, true);
-		if (err) {
-			dev_warn(rvu->dev, "SSO_HWGRP(%d)_AW_STATUS[XAQ_BUF_CACHED] not cleared",
-				 lf);
-			return err;
-		}
-
 		rvu_write64(rvu, blkaddr, SSO_AF_HWGRPX_XAQ_AURA(lf),
 			    npa_aura_id);
 		rvu_write64(rvu, blkaddr, SSO_AF_XAQX_GMCTL(lf),
-- 
2.17.1

